src/number_theory/bernoulli.lean

```diff
@@ -258,3 +265,35 @@ begin
   { ext x, rw bernoulli_eq_bernoulli' (succ_ne_zero x ∘ succ.inj) },
   { ring },
 end
+
+theorem bernoulli_power_series :
+  power_series.mk (λ n, (bernoulli n / n! : ℚ)) * (exp ℚ - 1) = X :=
+begin
+    suffices f : eval_neg_hom (power_series.mk (λ n, (bernoulli n / n! : ℚ)) *
+    (exp ℚ - 1)) = eval_neg_hom X,
+  { suffices g : function.injective eval_neg_hom,
+    { rwa g.eq_iff at f, },
+    apply rescale_injective,
+    norm_num, },
+  simp only [map_one, map_mul, eval_neg_hom_X, map_sub],
+  suffices h :
+    eval_neg_hom (mk (λ (n : ℕ), bernoulli n / ↑n!)) * (eval_neg_hom (exp ℚ) - 1) * (- exp ℚ)
+    = -X * (- exp ℚ),
+  { have hexp : - exp ℚ ≠ 0,
+    { simp only [exp, power_series.ext_iff, linear_map.map_zero, one_div, coeff_mk, coeff_one,
+        ring_hom.id_apply, linear_map.map_sub, ne.def, not_forall,
+        rat.algebra_map_rat_rat],
+      use 1,
+      simp only [factorial_one, coeff_mk, linear_map.map_neg, cast_one, inv_one, not_false_iff,
+        neg_eq_zero, one_ne_zero], },
+    apply mul_right_cancel' hexp h, },
+  { rw [mul_assoc],
+    have he : (eval_neg_hom (exp ℚ) - 1) * (- exp ℚ) = ((exp ℚ) - 1),
+    { ring,
+      simp only [exp_mul_exp_neg_eq_one],
+      ring, },
```

Comment:
The first `ring` here actually makes things more complicated for you. Try:
```suggestion
    have he : (eval_neg_hom (exp ℚ) - 1) * -exp ℚ = (exp ℚ) - 1,
    { simp [sub_mul, mul_comm (eval_neg_hom (exp ℚ)), exp_mul_exp_neg_eq_one] },
```
(I also removed some unnecessary parentheses.)